tpmutil/run_windows.go

```diff
@@ -0,0 +1,172 @@
+	flags   uint32
+}
+
+const (
+	tpm2Version            uint32  = 2
+	bothTPMVersionsFlag    uint32  = 6
+	tbsCommandLocalityZero uintptr = 0
+)
+
+// CommandPriority Parameter Docs
+// https://docs.microsoft.com/en-us/windows/desktop/api/Tbs/nf-tbs-tbsip_submit_command#parameters
+type CommandPriority uint32
+
+// CommandPriority is used to determine which pending command to submit whenever the TPM is free.
+// https://docs.microsoft.com/en-us/windows/desktop/tbs/command-scheduling
+const (
+	LowPriority     CommandPriority = 100          // used for low priority application use.
+	NormalPriority  CommandPriority = 200          // used for normal priority application use.
+	HighPriority    CommandPriority = 300          // used for high priority application use.
+	SystemPriority  CommandPriority = 400          // used for system tasks that access the TPM.
+	DefaultPriority                 = HighPriority // uses HighPriority as default
+)
+
+// Error Codes:
+// https://docs.microsoft.com/en-us/windows/desktop/TBS/tbs-return-codes
+var errMap = map[uintptr]string{
+	0x80284001: "An internal software error occurred.",
+	0x80284002: "One or more parameter values are not valid.",
+	0x80284003: "A specified output pointer is bad.",
+	0x80284004: "The specified context handle does not refer to a valid context.",
+	0x80284005: "The specified output buffer is too small.",
+	0x80284006: "An error occurred while communicating with the TPM.",
+	0x80284007: "A context parameter that is not valid was passed when attempting to create a TBS context.",
+	0x80284008: "The TBS service is not running and could not be started.",
+	0x80284009: "A new context could not be created because there are too many open contexts.",
+	0x8028400A: "A new virtual resource could not be created because there are too many open virtual resources.",
+	0x8028400B: "The TBS service has been started but is not yet running.",
+	0x8028400C: "The physical presence interface is not supported.",
+	0x8028400D: "The command was canceled.",
+	0x8028400E: "The input or output buffer is too large.",
+	0x8028400F: "A compatible Trusted Platform Module (TPM) Security Device cannot be found on this computer.",
+	0x80284010: "The TBS service has been disabled.",
+	0x80284011: "The TBS event log is not available.",
+	0x80284012: "The caller does not have the appropriate rights to perform the requested operation.",
+	0x80284013: "The TPM provisioning action is not allowed by the specified flags.",
+	0x80284014: "The Physical Presence Interface of this firmware does not support the requested method.",
+	0x80284015: "The requested TPM OwnerAuth value was not found.",
+}
+
+func tbsError(err uintptr) error {
+	if err == 0 {
+		return nil
+	}
+	if description, ok := errMap[err]; ok {
+		return fmt.Errorf("TBS Error: %s", description)
+	}
+	return fmt.Errorf("TBS Error: %v", err)
+}
+
+// winTPMBuffer is a ReadWriteCloser to access the TPM in Windows.
+type winTPMBuffer struct {
+	hContext  uintptr
```

Comment:
I realize that this is probably `hContext` in the API, but Go style does not use this kind of prefix notation. So, I'd prefer this to be `context` rather than `hContext`

Reply:
Done